### PR TITLE
Solenoid controller for SAM

### PR DESCRIPTION
**Pull request by aneedalie**
I intended on changing `actuatorsetup.cpp` to be used for the serial port initialization here as well, but I keep getting UI errors which took up most of my time, so I'm only including only the necessary changes to get keyboard controls moving for now.

Also started a small design doc on what I intend on doing, mostly for my purposes but others might find this helpful too: https://docs.google.com/document/d/1k3Tj2MtkHR4UH4iOGEJI9MBc-fQuhbIg326zg5GY4cg/edit?usp=sharing

### CODE REVIEW

File: code/controller/solenoid.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#include "solenoid.h"
+#include <ostream>
+
+Solenoid::Solenoid(const QString& serial_port): Controller(1, 1) {
+    // TODO: change actuator setup to be used for general serial set up
+    if ( serial_port.isEmpty() ) {
+        // TODO: provide error if this does not succeed
+        stream.open("/dev/ttyACM0");        // manually changed to match port indicated in arduino IDE
```
Comment (Mogball) [inline code/controller/solenoid.cpp]: it should be possible to autodetect ports if not provided

File: code/controller/controller.h [hunk 2]
```diff
@@ -19,6 +19,7 @@ class Controller {
     enum Type {
         ACTUATOR,
         SIMULATOR,
+        SOLENOID,
```
Comment (Mogball) [inline code/controller/controller.h]: I think we can just ditch `ACTUATOR` then?

Comment (aneedalie) [reply code/controller/controller.h]: Yeah I want to slowly remove all the actuator files for now. Trying to remove everything at once proved to open a nest of bugs 😓

File: code/controller/solenoid.cpp (hunk 1 above)
Comment (aneedalie) [reply code/controller/solenoid.cpp]: We currently have that set up in `actuatorsetup.cpp` but we learned that it doesn't properly detect the arduino USB port.

**Pull request merged**